src/Microsoft.Extensions.DiagnosticAdapter/Internal/ProxyTypeEmitter.cs

```diff
@@ -260,6 +260,9 @@ private static bool VerifyProxySupport(ProxyBuilderContext context, Tuple<Type,
             il.Emit(OpCodes.Ldarg_0);
             il.Emit(OpCodes.Ldarg_1);
             il.Emit(OpCodes.Ldtoken, proxyType);
+
+            var getTypeFromHandle = typeof(Type).GetMethod("GetTypeFromHandle", BindingFlags.Static | BindingFlags.Public);
+            il.EmitCall(OpCodes.Call, getTypeFromHandle, null);
```

Comment:
Yes. I had assumed there had to be some cleaner way to do it too, but that's what it put out.